### PR TITLE
Add bulk operation annotatable id validation for bulk media and objects creation models

### PR DESCRIPTION
**Pull request by katia-qm**
Add bulk operation annotatable id validation for bulk media and objects creation models

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+### Internal
+
+- added `bulk_operation_annotatable_id` presence validation for bulk media/media objects creation models [PR#36](https://github.com/quality-match/hari-client/pull/36)
+
 ## [2.0.1] - 24.10.2024
 
-- Updated create_subset endpoint used during object_category creation[PR#35](https://github.com/quality-match/hari-client/pull/35)
+### Internal
+
+- updated create_subset endpoint used during object_category creation [PR#35](https://github.com/quality-match/hari-client/pull/35)
```
Comment (Dens49) [inline CHANGELOG.md:11]: what's the update?

File: hari_client/models/models.py [hunk 2]
```diff
@@ -883,6 +898,17 @@ class MediaObjectCreate(BaseModel):
 class BulkMediaObjectCreate(MediaObjectCreate):
     bulk_operation_annotatable_id: str
 
+    @model_validator(mode="before")
+    @classmethod
+    def check_bulk_operation_annotatable_id_omitted(
+        cls, data: typing.Any
+    ) -> typing.Any:
+        if cls.__name__ == "HARIMediaObject":
+            return data
```
Comment (Dens49) [inline hari_client/models/models.py:919]: maybe overriding the validator in the child class could work

File: hari_client/client/errors.py [hunk 3]
```diff
@@ -84,3 +84,14 @@ def __init__(
         super().__init__(
             f"The valid length for the {param_name} parameter is: {minimum=}, {maximum=}, but actual length is {length=}"
         )
+
+
+class BulkOperationAnnotatableIdMissing(Exception):
+    def __init__(
+        self,
+        message="The 'bulk_operation_annotatable_id' field is missing. "
```
Comment (Dens49) [inline hari_client/client/errors.py:92]: nitpick: if you want the message to always be the same, there's no reason to make it overridable

File: CHANGELOG.md (hunk 1 above)
Comment (katia-qm) [reply CHANGELOG.md:11]: I just added it to Internal section, it was without any

**Review by Dens49 — APPROVED** [review]
:ok_hand: just a small nitpick left

File: hari_client/upload/hari_uploader.py [hunk 4]
```diff
@@ -32,6 +34,13 @@ class HARIMediaObject(models.BulkMediaObjectCreate):
     # of the hari api, but is needed to store which object category subset the media object should belong to.
     object_category_subset_name: str | None = pydantic.Field(default=None, exclude=True)
 
+    @model_validator(mode="before")
+    @classmethod
+    def check_bulk_operation_annotatable_id_omitted(
+        cls, data: typing.Any
+    ) -> typing.Any:
+        return data
```
[the remark this answers is not in the document]
Comment (katia-qm) [reply hari_client/upload/hari_uploader.py:42]: good idea

**Pull request merged**